beacon/sync/src/main/java/tech/pegasys/teku/beacon/sync/forward/multipeer/chains/SyncSourceFactory.java

```diff
@@ -36,11 +38,18 @@ public SyncSourceFactory(final AsyncRunner asyncRunner, final TimeProvider timeP
     this.timeProvider = timeProvider;
   }
 
-  public SyncSource getOrCreateSyncSource(final Eth2Peer peer) {
+  public SyncSource getOrCreateSyncSource(final Eth2Peer peer, final Spec spec) {
     // Limit request rate to just a little under what we'd accept
     final int maxBlocksPerMinute = MAX_BLOCKS_PER_MINUTE - SYNC_BATCH_SIZE.intValue() - 1;
+
+    final int maxBlobsPerBlock = spec.getMaxBlobsPerBlock().orElse(0);
+
     final int maxBlobSidecarsPerMinute =
-        MAX_BLOB_SIDECARS_PER_MINUTE - SYNC_BLOB_SIDECARS_SIZE.intValue() - 1;
+        MAX_REQUEST_BLOCKS_DENEB
+            .times(UInt64.valueOf(maxBlobsPerBlock))
+            .minusMinZero(SYNC_BLOB_SIDECARS_SIZE.intValue() - 1)
```

Comment:
I think we should make this SYNC_BATCH_SIZE.times(maxBlobsPerBlock).intValue() - 1. I have removed SYNC_BLOB_SIDECARS_SIZE entirely in my PR.

Reply:
Basically SYNC_BATCH_SIZE is what we use for syncing blocks and we will make request spanning same slots, but blobs sidecars are more than one per slot so can just multiply this value by the max blobs per block

Reply:
Sounds right to me. Done